proxy/api/src/lib.rs

```diff
@@ -21,6 +21,7 @@
     clippy::multiple_crate_versions,
     clippy::or_fun_call,
     clippy::shadow_reuse,
+    clippy::clippy::option_if_let_else,
```

Comment:
I’d like to highlight this change since it might warrant some discussion. I personally find [this lint](https://rust-lang.github.io/rust-clippy/master/index.html#option_if_let_else) gets in the way. It is arguable whether `Option::map_or_else` is more concise or more readable. But using `Option::map_or_else` we definitely quickly run into limitations if we use control flow statements like `break`, the `?` try operator, or `.await`.

Reply:
Legit!